utopia-remix/app/routes-test/internal.projects.$id.access.spec.ts

```diff
@@ -0,0 +1,87 @@
+jest.mock('@openfga/sdk')
+import { prisma } from '../db.server'
+import { handleChangeProjectAccess } from '../routes/internal.projects.$id.access'
+import {
+  clearDb,
+  createTestProject,
+  createTestSession,
+  createTestUser,
+  newTestRequest,
+} from '../test-util'
+import { ApiError } from '../util/errors'
+import * as permissionsService from '../services/permissionsService.server'
+import { AccessLevel } from '../types'
+
+describe('handleChangeAccess', () => {
```

Comment:
could you please also test the case where the user is not authenticated (since it's part of the endpoint logic)

Reply:
should we test it here? since the validator is actually the one who checks it (and now not the endpoint itself)

Reply:
the endpoint has a `requireUser` call which is checking authentication, not authorization, and imo the tests should reflect that. e.g. https://github.com/concrete-utopia/utopia/pull/4929#discussion_r1507942054 ← this should have had the tests fail